File: app/src/main/java/me/wangyuwei/xitukotlin/utils/AVUtils.java
package me.wangyuwei.xitukotlin.utils;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

import me.wangyuwei.xitukotlin.BuildConfig;

/**
 * 作者： 巴掌 on 16/4/24 14:17
 */
public class AVUtils {

    private static final ThreadLocal<SimpleDateFormat> THREAD_LOCAL_DATE_FORMAT = new ThreadLocal();
    public static final int TYPE_MOBILE = 2;
    public static final int TYPE_NOT_CONNECTED = 0;
    public static final int TYPE_WIFI = 1;
    private static final String dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";

    public static Date dateFromString(String str) {
        if (isBlankString(str)) {
            return null;
        }
        if (isDigitString(str)) {
            return new Date(Long.parseLong(str));
        }
        Date parse;
        SimpleDateFormat simpleDateFormat = (SimpleDateFormat) THREAD_LOCAL_DATE_FORMAT.get();
        if (simpleDateFormat == null) {
            simpleDateFormat = new SimpleDateFormat(dateFormat);
            simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
            THREAD_LOCAL_DATE_FORMAT.set(simpleDateFormat);
        }
        try {
            parse = simpleDateFormat.parse(str);
        } catch (Exception e) {
            parse = null;
        }
        return parse;
    }

    public static boolean isBlankString(String str) {
        return str == null || str.trim().equals(BuildConfig.FLAVOR);
    }

    public static boolean isDigitString(String str) {
        if (str == null) {
            return false;
        }
        for (int i = TYPE_NOT_CONNECTED; i < str.length(); i += TYPE_WIFI) {
            if (!Character.isDigit(str.charAt(i))) {
                return false;
            }
        }
        return true;
    }

}

File: app/src/main/java/me/wangyuwei/xitukotlin/utils/AVUtils.kt
package me.wangyuwei.xitukotlin.utils

import java.text.SimpleDateFormat
import java.util.Date
import java.util.TimeZone

import me.wangyuwei.xitukotlin.BuildConfig

/**
 * 作者： 巴掌 on 16/4/24 14:17
 */

val TYPE_MOBILE = 2
val TYPE_NOT_CONNECTED = 0
val TYPE_WIFI = 1
private val dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"

fun dateFromString(str: String): Date? {
    if (isBlankString(str)) {
        return null
    }
    if (isDigitString(str)) {
        return Date(java.lang.Long.parseLong(str))
    }

    val simpleDateFormat: SimpleDateFormat = SimpleDateFormat(dateFormat)
    simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"))

    val parse: Date?

    try {
        parse = simpleDateFormat.parse(str)
    } catch (e: Exception) {
        parse = null
    }

    return parse
}

fun isBlankString(str: String?): Boolean {
    return str == null || str.trim { it <= ' ' } == BuildConfig.FLAVOR
}

fun isDigitString(str: String?): Boolean {
    if (str == null) {
        return false
    }
    var i = TYPE_NOT_CONNECTED
    while (i < str.length) {
        if (!Character.isDigit(str[i])) {
            return false
        }
        i += TYPE_WIFI
    }
    return true
}

